src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.cs

```diff
@@ -184,8 +191,7 @@ private void AddToEventsMap_NoLock(ImmutableArray<CompilationEvent> compilationE
 
         private void UpdateEventsMap_NoLock(CompilationEvent compilationEvent, bool add)
         {
-            var symbolEvent = compilationEvent as SymbolDeclaredCompilationEvent;
-            if (symbolEvent != null)
+            if (compilationEvent is SymbolDeclaredCompilationEvent symbolEvent)
```

Comment:
Not really relevant to this PR, but I just want to make sure I'm understand the code: this can be rewritten as a switch, right?

Reply:
Yes, I can convert it to a switch in this PR.